v1/topdown/eval.go

```diff
@@ -61,54 +62,74 @@ type eval struct {
 	metrics                     metrics.Metrics
 	seed                        io.Reader
+	cancel                      Cancel
+	queryCompiler               ast.QueryCompiler
+	store                       storage.Store
+	txn                         storage.Transaction
+	virtualCache                VirtualCache
+	interQueryBuiltinCache      cache.InterQueryCache
+	interQueryBuiltinValueCache cache.InterQueryValueCache
+	printHook                   print.Hook
 	time                        *ast.Term
-	queryID                     uint64
 	queryIDFact                 *queryIDFactory
 	parent                      *eval
 	caller                      *eval
-	cancel                      Cancel
-	query                       ast.Body
-	queryCompiler               ast.QueryCompiler
-	index                       int
-	indexing                    bool
-	earlyExit                   bool
 	bindings                    *bindings
-	store                       storage.Store
 	baseCache                   *baseCache
-	txn                         storage.Transaction
 	compiler                    *ast.Compiler
 	input                       *ast.Term
 	data                        *ast.Term
 	external                    *resolverTrie
 	targetStack                 *refStack
-	tracers                     []QueryTracer
-	traceEnabled                bool
 	traceLastLocation           *ast.Location // Last location of a trace event.
-	plugTraceVars               bool
 	instr                       *Instrumentation
 	builtins                    map[string]*Builtin
 	builtinCache                builtins.Cache
 	ndBuiltinCache              builtins.NDBCache
 	functionMocks               *functionMocksStack
-	virtualCache                VirtualCache
 	comprehensionCache          *comprehensionCache
-	interQueryBuiltinCache      cache.InterQueryCache
-	interQueryBuiltinValueCache cache.InterQueryValueCache
 	saveSet                     *saveSet
 	saveStack                   *saveStack
 	saveSupport                 *saveSupport
 	saveNamespace               *ast.Term
-	skipSaveNamespace           bool
 	inliningControl             *inliningControl
-	genvarprefix                string
-	genvarid                    int
 	runtime                     *ast.Term
 	builtinErrors               *builtinErrors
-	printHook                   print.Hook
+	roundTripper                CustomizeRoundTripper
+	genvarprefix                string
+	query                       ast.Body
+	tracers                     []QueryTracer
 	tracingOpts                 tracing.Options
+	queryID                     uint64
+	index                       int
+	genvarid                    int
+	indexing                    bool
+	earlyExit                   bool
+	traceEnabled                bool
+	plugTraceVars               bool
+	skipSaveNamespace           bool
 	findOne                     bool
 	strictObjects               bool
-	roundTripper                CustomizeRoundTripper
+}
```

Comment:
I'm generally not a fan of ordering structs for their memory layout (that's what you did, isn't it?) but since this struct is so large, I don't think it becomes more unusable 💦

Reply:
Yeah me neither, but this one is huge, internal and instantiated all the the time… so I figured it would be one place where it’s worth it. I should add a comment here about that though so we make sure it’s kept that way even after future modifications.